### PR TITLE
[Kernel] Pipe attn_logits_soft_cap through paged attention TPU kernels

### PR DESCRIPTION
**Pull request by fenghuizhang**
Pipe attn_logits_soft_cap through paged_attention, this will unblock some of our models' adoption.

Note that the changed code currently doesn't have unit tests. I will add one later.

Signed-off-by: Fenghui Zhang <fhzhang@google.com>

### CODE REVIEW

**Review by robertgshaw2-redhat — APPROVED** [review]

**Comment by robertgshaw2-redhat** [comment]
Can you add a correctness test for gemma in the ci so this codepath is flexed? Thanks!

**Comment by mgoin** [comment]
Ditto, LGTM when we can add test case to CI

**Comment by fenghuizhang** [comment]
Local tests with the latest pt/xla code pass now, we may need to update the nightly build label for CI to pass:

```
PYTHONPATH=/home/fhzhang/xla python3 examples/offline_inference/tpu.py
```

The error in https://buildkite.com/vllm/fastcheck/builds/12156#01948bc4-58cd-4863-9eca-e2ea098879f9 went away.
